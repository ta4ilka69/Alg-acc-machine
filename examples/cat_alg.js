char c = input();
while (c){
    print(c);
    c = input();
}
