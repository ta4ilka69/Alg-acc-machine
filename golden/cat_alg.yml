in_source: |-
  char c = input();
  while (c){
      print(c);
      c = input();
  }
in_stdin: |-
  it's a cat
out_log: |
  DEBUG   machine:simulation    TICK:   0 PC:   0 AR: 0 DR: 0 ACC:0 	input
  DEBUG   machine:simulation    TICK:   1 PC:   1 AR: 0 DR: 0 ACC:105 	store 0
  DEBUG   machine:simulation    TICK:   2 PC:   2 AR: 0 DR: 0 ACC:105 	load 0
  DEBUG   machine:simulation    TICK:   3 PC:   3 AR: 0 DR: 105 ACC:105 	store 4095
  DEBUG   machine:simulation    TICK:   4 PC:   4 AR: 4095 DR: 105 ACC:105 	load 4095
  DEBUG   machine:simulation    TICK:   5 PC:   5 AR: 4095 DR: 105 ACC:105 	jz 13
  DEBUG   machine:simulation    TICK:   6 PC:   6 AR: 4095 DR: 105 ACC:105 	load 0
  DEBUG   machine:simulation    TICK:   7 PC:   7 AR: 0 DR: 105 ACC:105 	store 4095
  DEBUG   machine:simulation    TICK:   8 PC:   8 AR: 4095 DR: 105 ACC:105 	load 4095
  DEBUG   machine:simulation    TICK:   9 PC:   9 AR: 4095 DR: 105 ACC:105 	output
  DEBUG   machine:simulation    TICK:  10 PC:  10 AR: 4095 DR: 105 ACC:105 	input
  DEBUG   machine:simulation    TICK:  11 PC:  11 AR: 4095 DR: 105 ACC:116 	store 0
  DEBUG   machine:simulation    TICK:  12 PC:  12 AR: 0 DR: 105 ACC:116 	jump 2
  DEBUG   machine:simulation    TICK:  13 PC:   2 AR: 0 DR: 105 ACC:116 	load 0
  DEBUG   machine:simulation    TICK:  14 PC:   3 AR: 0 DR: 116 ACC:116 	store 4095
  DEBUG   machine:simulation    TICK:  15 PC:   4 AR: 4095 DR: 116 ACC:116 	load 4095
  DEBUG   machine:simulation    TICK:  16 PC:   5 AR: 4095 DR: 116 ACC:116 	jz 13
  DEBUG   machine:simulation    TICK:  17 PC:   6 AR: 4095 DR: 116 ACC:116 	load 0
  DEBUG   machine:simulation    TICK:  18 PC:   7 AR: 0 DR: 116 ACC:116 	store 4095
  DEBUG   machine:simulation    TICK:  19 PC:   8 AR: 4095 DR: 116 ACC:116 	load 4095
  DEBUG   machine:simulation    TICK:  20 PC:   9 AR: 4095 DR: 116 ACC:116 	output
  DEBUG   machine:simulation    TICK:  21 PC:  10 AR: 4095 DR: 116 ACC:116 	input
  DEBUG   machine:simulation    TICK:  22 PC:  11 AR: 4095 DR: 116 ACC:39 	store 0
  DEBUG   machine:simulation    TICK:  23 PC:  12 AR: 0 DR: 116 ACC:39 	jump 2
  DEBUG   machine:simulation    TICK:  24 PC:   2 AR: 0 DR: 116 ACC:39 	load 0
  DEBUG   machine:simulation    TICK:  25 PC:   3 AR: 0 DR: 39 ACC:39 	store 4095
  DEBUG   machine:simulation    TICK:  26 PC:   4 AR: 4095 DR: 39 ACC:39 	load 4095
  DEBUG   machine:simulation    TICK:  27 PC:   5 AR: 4095 DR: 39 ACC:39 	jz 13
  DEBUG   machine:simulation    TICK:  28 PC:   6 AR: 4095 DR: 39 ACC:39 	load 0
  DEBUG   machine:simulation    TICK:  29 PC:   7 AR: 0 DR: 39 ACC:39 	store 4095
  DEBUG   machine:simulation    TICK:  30 PC:   8 AR: 4095 DR: 39 ACC:39 	load 4095
  DEBUG   machine:simulation    TICK:  31 PC:   9 AR: 4095 DR: 39 ACC:39 	output
  DEBUG   machine:simulation    TICK:  32 PC:  10 AR: 4095 DR: 39 ACC:39 	input
  DEBUG   machine:simulation    TICK:  33 PC:  11 AR: 4095 DR: 39 ACC:115 	store 0
  DEBUG   machine:simulation    TICK:  34 PC:  12 AR: 0 DR: 39 ACC:115 	jump 2
  DEBUG   machine:simulation    TICK:  35 PC:   2 AR: 0 DR: 39 ACC:115 	load 0
  DEBUG   machine:simulation    TICK:  36 PC:   3 AR: 0 DR: 115 ACC:115 	store 4095
  DEBUG   machine:simulation    TICK:  37 PC:   4 AR: 4095 DR: 115 ACC:115 	load 4095
  DEBUG   machine:simulation    TICK:  38 PC:   5 AR: 4095 DR: 115 ACC:115 	jz 13
  DEBUG   machine:simulation    TICK:  39 PC:   6 AR: 4095 DR: 115 ACC:115 	load 0
  DEBUG   machine:simulation    TICK:  40 PC:   7 AR: 0 DR: 115 ACC:115 	store 4095
  DEBUG   machine:simulation    TICK:  41 PC:   8 AR: 4095 DR: 115 ACC:115 	load 4095
  DEBUG   machine:simulation    TICK:  42 PC:   9 AR: 4095 DR: 115 ACC:115 	output
  DEBUG   machine:simulation    TICK:  43 PC:  10 AR: 4095 DR: 115 ACC:115 	input
  DEBUG   machine:simulation    TICK:  44 PC:  11 AR: 4095 DR: 115 ACC:32 	store 0
  DEBUG   machine:simulation    TICK:  45 PC:  12 AR: 0 DR: 115 ACC:32 	jump 2
  DEBUG   machine:simulation    TICK:  46 PC:   2 AR: 0 DR: 115 ACC:32 	load 0
  DEBUG   machine:simulation    TICK:  47 PC:   3 AR: 0 DR: 32 ACC:32 	store 4095
  DEBUG   machine:simulation    TICK:  48 PC:   4 AR: 4095 DR: 32 ACC:32 	load 4095
  DEBUG   machine:simulation    TICK:  49 PC:   5 AR: 4095 DR: 32 ACC:32 	jz 13
  DEBUG   machine:simulation    TICK:  50 PC:   6 AR: 4095 DR: 32 ACC:32 	load 0
  DEBUG   machine:simulation    TICK:  51 PC:   7 AR: 0 DR: 32 ACC:32 	store 4095
  DEBUG   machine:simulation    TICK:  52 PC:   8 AR: 4095 DR: 32 ACC:32 	load 4095
  DEBUG   machine:simulation    TICK:  53 PC:   9 AR: 4095 DR: 32 ACC:32 	output
  DEBUG   machine:simulation    TICK:  54 PC:  10 AR: 4095 DR: 32 ACC:32 	input
  DEBUG   machine:simulation    TICK:  55 PC:  11 AR: 4095 DR: 32 ACC:97 	store 0
  DEBUG   machine:simulation    TICK:  56 PC:  12 AR: 0 DR: 32 ACC:97 	jump 2
  DEBUG   machine:simulation    TICK:  57 PC:   2 AR: 0 DR: 32 ACC:97 	load 0
  DEBUG   machine:simulation    TICK:  58 PC:   3 AR: 0 DR: 97 ACC:97 	store 4095
  DEBUG   machine:simulation    TICK:  59 PC:   4 AR: 4095 DR: 97 ACC:97 	load 4095
  DEBUG   machine:simulation    TICK:  60 PC:   5 AR: 4095 DR: 97 ACC:97 	jz 13
  DEBUG   machine:simulation    TICK:  61 PC:   6 AR: 4095 DR: 97 ACC:97 	load 0
  DEBUG   machine:simulation    TICK:  62 PC:   7 AR: 0 DR: 97 ACC:97 	store 4095
  DEBUG   machine:simulation    TICK:  63 PC:   8 AR: 4095 DR: 97 ACC:97 	load 4095
  DEBUG   machine:simulation    TICK:  64 PC:   9 AR: 4095 DR: 97 ACC:97 	output
  DEBUG   machine:simulation    TICK:  65 PC:  10 AR: 4095 DR: 97 ACC:97 	input
  DEBUG   machine:simulation    TICK:  66 PC:  11 AR: 4095 DR: 97 ACC:32 	store 0
  DEBUG   machine:simulation    TICK:  67 PC:  12 AR: 0 DR: 97 ACC:32 	jump 2
  DEBUG   machine:simulation    TICK:  68 PC:   2 AR: 0 DR: 97 ACC:32 	load 0
  DEBUG   machine:simulation    TICK:  69 PC:   3 AR: 0 DR: 32 ACC:32 	store 4095
  DEBUG   machine:simulation    TICK:  70 PC:   4 AR: 4095 DR: 32 ACC:32 	load 4095
  DEBUG   machine:simulation    TICK:  71 PC:   5 AR: 4095 DR: 32 ACC:32 	jz 13
  DEBUG   machine:simulation    TICK:  72 PC:   6 AR: 4095 DR: 32 ACC:32 	load 0
  DEBUG   machine:simulation    TICK:  73 PC:   7 AR: 0 DR: 32 ACC:32 	store 4095
  DEBUG   machine:simulation    TICK:  74 PC:   8 AR: 4095 DR: 32 ACC:32 	load 4095
  DEBUG   machine:simulation    TICK:  75 PC:   9 AR: 4095 DR: 32 ACC:32 	output
  DEBUG   machine:simulation    TICK:  76 PC:  10 AR: 4095 DR: 32 ACC:32 	input
  DEBUG   machine:simulation    TICK:  77 PC:  11 AR: 4095 DR: 32 ACC:99 	store 0
  DEBUG   machine:simulation    TICK:  78 PC:  12 AR: 0 DR: 32 ACC:99 	jump 2
  DEBUG   machine:simulation    TICK:  79 PC:   2 AR: 0 DR: 32 ACC:99 	load 0
  DEBUG   machine:simulation    TICK:  80 PC:   3 AR: 0 DR: 99 ACC:99 	store 4095
  DEBUG   machine:simulation    TICK:  81 PC:   4 AR: 4095 DR: 99 ACC:99 	load 4095
  DEBUG   machine:simulation    TICK:  82 PC:   5 AR: 4095 DR: 99 ACC:99 	jz 13
  DEBUG   machine:simulation    TICK:  83 PC:   6 AR: 4095 DR: 99 ACC:99 	load 0
  DEBUG   machine:simulation    TICK:  84 PC:   7 AR: 0 DR: 99 ACC:99 	store 4095
  DEBUG   machine:simulation    TICK:  85 PC:   8 AR: 4095 DR: 99 ACC:99 	load 4095
  DEBUG   machine:simulation    TICK:  86 PC:   9 AR: 4095 DR: 99 ACC:99 	output
  DEBUG   machine:simulation    TICK:  87 PC:  10 AR: 4095 DR: 99 ACC:99 	input
  DEBUG   machine:simulation    TICK:  88 PC:  11 AR: 4095 DR: 99 ACC:97 	store 0
  DEBUG   machine:simulation    TICK:  89 PC:  12 AR: 0 DR: 99 ACC:97 	jump 2
  DEBUG   machine:simulation    TICK:  90 PC:   2 AR: 0 DR: 99 ACC:97 	load 0
  DEBUG   machine:simulation    TICK:  91 PC:   3 AR: 0 DR: 97 ACC:97 	store 4095
  DEBUG   machine:simulation    TICK:  92 PC:   4 AR: 4095 DR: 97 ACC:97 	load 4095
  DEBUG   machine:simulation    TICK:  93 PC:   5 AR: 4095 DR: 97 ACC:97 	jz 13
  DEBUG   machine:simulation    TICK:  94 PC:   6 AR: 4095 DR: 97 ACC:97 	load 0
  DEBUG   machine:simulation    TICK:  95 PC:   7 AR: 0 DR: 97 ACC:97 	store 4095
  DEBUG   machine:simulation    TICK:  96 PC:   8 AR: 4095 DR: 97 ACC:97 	load 4095
  DEBUG   machine:simulation    TICK:  97 PC:   9 AR: 4095 DR: 97 ACC:97 	output
  DEBUG   machine:simulation    TICK:  98 PC:  10 AR: 4095 DR: 97 ACC:97 	input
  DEBUG   machine:simulation    TICK:  99 PC:  11 AR: 4095 DR: 97 ACC:116 	store 0
  DEBUG   machine:simulation    TICK: 100 PC:  12 AR: 0 DR: 97 ACC:116 	jump 2
  DEBUG   machine:simulation    TICK: 101 PC:   2 AR: 0 DR: 97 ACC:116 	load 0
  DEBUG   machine:simulation    TICK: 102 PC:   3 AR: 0 DR: 116 ACC:116 	store 4095
  DEBUG   machine:simulation    TICK: 103 PC:   4 AR: 4095 DR: 116 ACC:116 	load 4095
  DEBUG   machine:simulation    TICK: 104 PC:   5 AR: 4095 DR: 116 ACC:116 	jz 13
  DEBUG   machine:simulation    TICK: 105 PC:   6 AR: 4095 DR: 116 ACC:116 	load 0
  DEBUG   machine:simulation    TICK: 106 PC:   7 AR: 0 DR: 116 ACC:116 	store 4095
  DEBUG   machine:simulation    TICK: 107 PC:   8 AR: 4095 DR: 116 ACC:116 	load 4095
  DEBUG   machine:simulation    TICK: 108 PC:   9 AR: 4095 DR: 116 ACC:116 	output
  DEBUG   machine:simulation    TICK: 109 PC:  10 AR: 4095 DR: 116 ACC:116 	input
  WARNING machine:simulation    Input buffer is empty
  INFO    machine:simulation    output_buffer: "it's a cat"
out_stdout: |
  source LoC: 5 code instr: 14
  ============================================================
  it's a cat
  instr_counter:  109 ticks: 109
out_code: |
  {"opcode": "input", "index": 0}
  {"opcode": "store", "arg": 0, "index": 1}
  {"opcode": "load", "arg": 0, "addressing_mode": "absolute", "index": 2}
  {"opcode": "store", "arg": 4095, "index": 3}
  {"opcode": "load", "arg": 4095, "addressing_mode": "absolute", "index": 4}
  {"opcode": "jz", "index": 5, "arg": 13}
  {"opcode": "load", "arg": 0, "addressing_mode": "absolute", "index": 6}
  {"opcode": "store", "arg": 4095, "index": 7}
  {"opcode": "load", "arg": 4095, "addressing_mode": "absolute", "index": 8}
  {"opcode": "output", "index": 9}
  {"opcode": "input", "index": 10}
  {"opcode": "store", "arg": 0, "index": 11}
  {"opcode": "jump", "arg": 2, "index": 12}
  {"opcode": "halt", "index": 13}

  {"data": []}
